mono/metadata/appdomain.c

```diff
@@ -2336,9 +2341,14 @@ mono_domain_assembly_search (MonoAssemblyLoadContext *alc, MonoAssembly *request
 	for (tmp = domain->domain_assemblies; tmp; tmp = tmp->next) {
 		ass = (MonoAssembly *)tmp->data;
 		g_assert (ass != NULL);
+
+		MonoAssemblyName *overriden_name = (MonoAssemblyName *)g_hash_table_lookup (domain->overriden_names, ass);
+		gboolean name_matches = overriden_name ?
+			mono_assembly_names_equal_flags (aname, overriden_name, eq_flags) :	mono_assembly_names_equal_flags (aname, &ass->aname, eq_flags);
+
```

Comment:
Why does this hash table map assembly to name and not the other way around?

If we do it this way, I think we need a list of names.

Reply:
I guess probably the same reason there's a loop here, ugh.

Reply:
I think we need to store a list.